src/webapi/geolocation.rs

```diff
@@ -0,0 +1,135 @@
+//! Implementation of Geolocation API as defined at:
+//! https://developer.mozilla.org/en-US/docs/Web/API/Geolocation
+//! https://developer.mozilla.org/en-US/docs/Web/API/Geolocation/Using_geolocation
+//!
+//! Note that this implementation currently doesn't handle the failure to get position callbacks
+//! nor does it handle geo_options.
+
+use webcore::value::{Reference, Value};
+use webcore::try_from::TryInto;
+
+/// Representation of positional coordinate information.
+#[derive(Clone, Debug, ReferenceType)]
+#[reference(instance_of = "Object")]
+pub struct Coordinates(Reference);
```

Comment:
Hmm... I'd probably prefer if `Coordinates` and `Position` were normal Rust `struct`s instead of `Reference`s.

Reply:
They're reference right now because they're wrapping over values returned from the Javascript API. Should I take the values from the Javascript API and reencode them into Rust and pass that to the user?

Reply:
Yep. In this case I don't really see much point in keeping these as references.